File: 2107-find-unique-binary-string/find-unique-binary-string.java
class Solution {
    public String findDifferentBinaryString(String[] nums) {
        final int N = nums.length;

        Set<Integer> seen = new HashSet<>();
        for (int i = 0; i < N; i++) {
            seen.add(Integer.parseInt(nums[i], 2));
        }

        for (int i = 0, len = 1 << N; i < len; i++) {
            if (!seen.contains(i)) {
                String result = Integer.toBinaryString(i);
                return String.format("%" + N +  "s", result).replace(' ', '0');
            }
        }

        return "";
    }
}

File: 2107-find-unique-binary-string/find-unique-binary-string.cpp
class Solution {
public:
    string findDifferentBinaryString(vector<string>& nums) {
        const int N = nums.size();

        set<int> seen;
        for (int i = 0; i < N; i++) {
            seen.insert(stoi(nums[i], nullptr, 2));
        }

        for (int i = 0, len = 1 << N; i < len; i++) {
            if (seen.find(i) == seen.end()) {
                string result = bitset<16>(i).to_string();
                return result.substr(16 - N, N);
            }
        }

        return "";
    }
};